connector_onpay/wizards/onpay_data_export.py

```diff
@@ -13,179 +13,206 @@
 
 
 class OnPayExport(models.TransientModel):
-    _name = 'onpay.export'
+    _name = "onpay.export"
     _description = "OnPay Export Wizard"
 
-    date_from = fields.Date(
-        string='From'
-    )
-    date_to = fields.Date(
-        string='To'
-    )
+    date_from = fields.Date(string="From")
+    date_to = fields.Date(string="To")
     file_type = fields.Selection(
-        [('csv', 'CSV'),
-         ('xls', 'XLS')], default='csv', string="File Type")
+        [("csv", "CSV"), ("xls", "XLS")], default="csv", string="File Type"
+    )
 
     @api.multi
     def export_data(self):
-        if self.file_type == 'csv':
+        if self.file_type == "csv":
             res_id, context = self.export_data_csv()
         else:
             res_id, context = self.export_data_xls()
         return {
-            'type': 'ir.actions.act_window',
-            'view_type': 'form',
-            'view_mode': 'form',
-            'res_model': 'wiz.xls.content',
-            'target': 'new',
-            'res_id': res_id.id,
-            'context': context,
+            "type": "ir.actions.act_window",
+            "view_type": "form",
+            "view_mode": "form",
+            "res_model": "wiz.xls.content",
+            "target": "new",
+            "res_id": res_id.id,
+            "context": context,
         }
 
     @api.multi
     def employee_timesheet_data(self, date_from, date_to):
         # Timesheet data of employees between selected date
-        self._cr.execute("""select employee_id,line_onpay_id,
-                sum(unit_amount) as unit_amount from
-                account_analytic_line where
+        self._cr.execute(
+            """select aal.employee_id, aal.line_onpay_id,
+                sum(aal.unit_amount) as unit_amount from
+                account_analytic_line as aal
+                LEFT JOIN hr_employee he ON aal.employee_id = he.id
+                where he.onpay_code IS NOT Null and
                 date >= %s and date <= %s and employee_id in %s group by
```

Comment:
We are missing the filter for time_type onpay_code is not null

Reply:
In a new code, we required a field in a view level

Reply:
onpay_code is optional, and we want to filter out empty values